.github/workflows/int-and-connected-test-run.yml

```diff
@@ -108,100 +132,81 @@ jobs:
           TDR_LOG_APPENDER: 'Console-Standard'
-        with:
-          actions_subcommand: 'gradleinttest'
-          pgport: ${{ job.services.postgres.ports[5432] }}
-          test_to_run: 'testIntegration'
-          sa_b64_credentials: ${{ secrets.SA_B64_CREDENTIALS }}
-      - name: Clean state lock from used Namespace on API deploy
-        if: always()
-        uses: broadinstitute/datarepo-actions/actions/main@0.74.0
-        with:
-          actions_subcommand: 'k8_checknamespace_clean'
-          sa_b64_credentials: ${{ secrets.SA_B64_CREDENTIALS }}
-      - name: Clean whitelisted Runner IP
+        run: |
+          # wait for postgres to be ready
+          pg_isready -h ${PGHOST} -t 30
+          # create the datarepo and stairway databases
+          psql -U postgres -f ./db/create-data-repo-db
+          # extract service account credentials
+          base64 --decode <<< ${{ secrets.SA_B64_CREDENTIALS }} > ${GOOGLE_APPLICATION_CREDENTIALS}
+          jq -r .private_key ${GOOGLE_APPLICATION_CREDENTIALS} > ${GOOGLE_SA_CERT}
+          chmod 644 ${GOOGLE_SA_CERT}
+          base64 --decode <<< ${{ secrets.B64_RBS_APPLICATION_CREDENTIALS }} > ${RBS_CLIENT_CREDENTIAL_FILE_PATH}
+          # build code and run local api
+          ./gradlew bootJar
+          export DATA_REPO_JAR=$(find . -type f -name jade-data-repo-*-SNAPSHOT.jar)
+          java -jar ${DATA_REPO_JAR} > local-api-output.log &
+          # wait until api is ready
+          timeout 30 bash -c 'until curl -s ${IT_JADE_API_URL}/status; do sleep 1; done'
+          # run integration tests
+          ./gradlew --scan testIntegration
+      - name: Upload API logs
         if: always()
-        uses: broadinstitute/datarepo-actions/actions/main@0.74.0
+        uses: actions/upload-artifact@v4
         with:
-          actions_subcommand: 'gcp_whitelist_clean'
-          sa_b64_credentials: ${{ secrets.SA_B64_CREDENTIALS }}
+          name: api-logs
+          path: local-api-output.log
+          retention-days: 7
+  git_hash:
+    name: Extract git hash
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.config.outputs.GIT_HASH }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Get the latest git hash
+        id: config
+        run: |
+          GIT_HASH=$(git rev-parse --short HEAD)
+          echo "GIT_HASH=${GIT_HASH}" >> $GITHUB_OUTPUT
+          echo "Latest git hash in branch is ${GIT_HASH}"
   report-to-sherlock:
-    uses: broadinstitute/sherlock/.github/workflows/client-report-app-version.yaml@main
-    needs: test_integration
-    # Always attempt to run if pull_request, as we want to report the appVersion even if the tests fail.
-    # never run on cron or other runs as we don't want extranaeous build reporting.
     if: github.event_name == 'pull_request'
+    uses: broadinstitute/sherlock/.github/workflows/client-report-app-version.yaml@main
+    needs: git_hash
```

Comment:
A happy consequence of this particular change: this means that a dev can push changes to their PR and very soon after be able to deploy the feature branch to their BEE!  Previously we had to wait at the end of our very long integration test run.

From https://broadworkbench.atlassian.net/browse/DCJ-38 --

>We can follow examples in other repos to build the tdr image and report the hash to sherlock. 
>
>We already do this in our int test action, but it happens after the tests complete (which takes ~2.5 hours!!).

I'd consider closing that ticket if we feel like this is an acceptable interpretation.

Reply:
I agree, this is nice, but I'm not sure if this is the correct behavior for `dev` builds. Could this result in a version being reported to sherlock for dev that doesn't pass tests, after the PR was merged up?

Reply:
Are you referring to the `report-to-sherlock`, the `report-workflow` job, or both? From the GitHub docs:

> By default, a workflow only runs when a pull_request event's activity type is opened, synchronize, or reopened

Also, the previous comment said:

> Always attempt to run if pull_request, as we want to report the appVersion **even if the tests fail**.
> never run on cron or other runs as we don't want extranaeous build reporting.

So I think `report-to-sherlock` will not run on `dev` builds. For `report-workflow`, the job did not depend on tests, and does not depend on tests now.

Reply:
I agree with @fboulnois -- this `report-to-sherlock` task only runs on PRs, per line 197 above.
```
    if: github.event_name == 'pull_request'
```

Reply:
OK now I see, thanks for the explanation. Also I didn't see that report-workflow waits for the entire workflow to complete, so it doesn't matter that it's called inside a job that doesn't depend on the test results.